File: code_examples/3-Structured_Problem_Solving/7-If_Then_Else/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program checks if ther is over 30 students

#include <stdio.h>

int main() {
    // this function checks if ther is over 30 students
    const int MAX_STUDENT_NUMBER = 30;
    int numberOfStudents;

    // input
    printf("Enter the number of students: ");
    scanf("%d", &numberOfStudents);

    // process
    if (numberOfStudents > MAX_STUDENT_NUMBER) {
        // output
        printf("Too many students!\n");
    }

    printf("\nDone.\n");
    return 0;
}

File: code_examples/3-Structured_Problem_Solving/7-If_Then_Else/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module checks if there is over 30 students
"""


from constants import MAX_STUDENT_NUMBER


def main() -> None:
    """The main() function checks if over 30 students, returns None."""

    # input
    number_of_students = int(input("Enter the number of students: "))

    # process and output
    if number_of_students > MAX_STUDENT_NUMBER:
        print("Too many students!")
    else:
        print("Not too many students.")

    print("\nDone.")


if __name__ == "__main__":
    main()
